streampark-common/src/main/scala/org/apache/streampark/common/util/CompletableFutureUtils.scala

```diff
@@ -75,36 +80,47 @@ object CompletableFutureUtils {
       unit: TimeUnit,
       handle: Consumer[T],
       exceptionally: Consumer[Throwable]): CompletableFuture[Unit] = {
-    future.applyToEither(
+
+    val unitFuture = future.applyToEither(
       setTimeout(timeout, unit),
-      new JavaFunc[T, Unit] {
+      new JavaFunc[T, Unit]() {
         override def apply(t: T): Unit = {
           if (handle != null) {
             handle.accept(t)
           }
         }
-      }).exceptionally(new JavaFunc[Throwable, Unit] {
+      }).exceptionally(new JavaFunc[Throwable, Unit]() {
       override def apply(t: Throwable): Unit = {
         if (exceptionally != null) {
           exceptionally.accept(t)
         }
       }
     })
+    cancelUnitFuture(unitFuture)
   }
 
   def runTimeout[T](future: CompletableFuture[T], timeout: Long, unit: TimeUnit): CompletableFuture[Unit] = {
-    runTimeout(
+    val unitFuture = runTimeout(
       future,
       timeout,
       unit,
       null,
-      new Consumer[Throwable] {
+      new Consumer[Throwable]() {
         override def accept(t: Throwable): Unit = {
           if (!future.isDone) {
             future.cancel(true)
           }
         }
       })
+    cancelUnitFuture(unitFuture)
```

Comment:
The unitFuture is canceled three times:

<img width="881" alt="image" src="https://user-images.githubusercontent.com/38427477/211469510-d7627789-c893-4067-bccb-6d199667db24.png">